micromanager_gui/_gui_objects/_group_preset_table_widget.py

```diff
@@ -0,0 +1,120 @@
+from typing import Any
+
+from magicgui.widgets import ComboBox, FloatSlider, LineEdit, Table, Widget
+from qtpy import QtWidgets as QtW
+from qtpy.QtWidgets import QVBoxLayout
+
+from .. import _core
+
+
+class MainTable(Table):
+    def __init__(self) -> None:
+        super().__init__()
+        hdr = self.native.horizontalHeader()
+        hdr.setSectionResizeMode(hdr.Stretch)
+        vh = self.native.verticalHeader()
+        vh.setVisible(False)
+        vh.setSectionResizeMode(vh.Fixed)
+        vh.setDefaultSectionSize(24)
+        self.native.setEditTriggers(QtW.QTableWidget.NoEditTriggers)
+
+
+class MMGroupPresetTableWidget(QtW.QWidget):
+    def __init__(self):
+        super().__init__()
+
+        self._mmc = _core.get_core_singleton()
+        self._mmc.events.systemConfigurationLoaded.connect(self._populate_table)
+
+        self.table_wdg = MainTable()
+        self.table_wdg.column_headers = ("Groups", "Presets")
+        self.table_wdg.show()
+        self.setLayout(QVBoxLayout())
+        self.setContentsMargins(0, 0, 0, 0)
+        self.layout().addWidget(self.table_wdg.native)
+
+    def _on_system_cfg_loaded(self):
+        self._populate_table()
+
+    def _populate_table(self):
+        self.table_wdg.clear()
+        groups = self._mmc.getAvailableConfigGroups()
+        if not groups:
+            return
+        data = []
+        for group in groups:
+            presets = list(self._mmc.getAvailableConfigs(group))
+            wdg = self._set_widget(group, presets)
+            data.append([group, wdg])
+        self.table_wdg.value = {
+            "data": data,
+            "index": [],
+            "columns": ["Groups", "Presets"],
+        }
+
+    def _get_cfg_data(self, group, preset):
+        for n, key in enumerate(self._mmc.getConfigData(group, preset)):
+            dev = key[0]
+            prop = key[1]
+            val = key[2]
+        return dev, prop, val, (n + 1)
+
+    def _set_float_slider_wdg(self, presets, device, property, value):
+        return FloatSlider(
+            value=float(value),
+            min=float(self._mmc.getPropertyLowerLimit(device, property)),
+            max=float(self._mmc.getPropertyUpperLimit(device, property)),
+            label=str(property),
+            name=f"{presets[0]}",
+            annotation=[device, property, value],
+        )
+
+    def _set_widget(self, group, presets) -> Widget:
+        wdg = None
+
+        device, property, value, count = self._get_cfg_data(group, presets[0])
+
+        if len(presets) > 1:
+            wdg = ComboBox(choices=presets, name=f"{presets}", annotation=[])
```

Comment:
that's not really what annotation is for... it should be a type annotation  (but can generally be omitted).  Are you using it to store info or something?

Reply:
We can get rid of it at the moment. I was thinking to use it for something later on but probably it won't be necessary.